client/state/imports/reducer.js

```diff
@@ -0,0 +1,290 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { flow, curry, omit, omitBy, includes, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	IMPORTS_STORE_RESET,
+	IMPORTS_FETCH,
+	IMPORTS_FETCH_FAILED,
+	IMPORTS_FETCH_COMPLETED,
+	IMPORTS_IMPORT_START,
+	IMPORTS_START_IMPORTING,
+	IMPORTS_IMPORT_CANCEL,
+	IMPORTS_IMPORT_RESET,
+	IMPORTS_IMPORT_RECEIVE,
+	IMPORTS_UPLOAD_START,
+	IMPORTS_UPLOAD_FAILED,
+	IMPORTS_UPLOAD_SET_PROGRESS,
+	IMPORTS_UPLOAD_COMPLETED,
+	IMPORTS_AUTHORS_START_MAPPING,
+	IMPORTS_AUTHORS_SET_MAPPING,
+	IMPORTS_IMPORT_LOCK,
+	IMPORTS_IMPORT_UNLOCK,
+} from 'state/action-types';
+import { appStates } from 'state/imports/constants';
+
+/**
+ * The inital state of the imports store
+ *
+ * @type {Object}
+ */
+const initialState = {
+	count: 0,
+	importers: {},
+	importerLocks: {},
+	api: {
+		isHydrated: false,
+		isFetching: false,
+		retryCount: 0,
+	},
+};
+
+/**
+ * Increment the count property of the store state.
+ *
+ * @param {Object} state The state of the redux store
+ * @returns {Object} state The updated state with count incremented by 1
+ */
+const incrementCount = state => ( { ...state, count: state.count + 1 } );
+
+/**
+ * Merge an update into the api property of the store.
+ *
+ * @param {Object} apiUpdate The object to merge into the api object
+ * @returns {Object} The updated state with api changes
+ */
+const mergeApiUpdate = curry( ( apiUpdate, state ) => ( {
```

Comment:
I think we'd like to avoid the currying syntax here. This is some pretty old code and I think we'll want to consider updating it instead of just moving it over into Redux, and hopefully that's not a massive chore switching paradigms.

That is, it would be nice to see the functions rewritten without `curry()`, preferring `apiUpdate => state => …` syntax or even better, uncurried with `( state, apiUpdate )` args, for example

Reply:
👍 will make those changes